### DISCUSSION TITLE
NavigationLink in iOS17 usage

### DISCUSSION THREAD

**Discussion by tigline** [pointfreeco/swift-composable-architecture#2751] [General]
### Description

In the example SwiftCaseStudies-> "NavigateAndLoadList"

the NavigationLink get the **warning** :**'init(_:tag:selection:destination:)' was deprecated in iOS 16.0: use NavigationLink(value:label:), or navigationDestination(isPresented:destination:), inside a NavigationStack or NavigationSplitView**

I will use iOS 17 as base version. So when will the example fix to adapter to iOS 17. I have try but failed.

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

NavigationLink in example SwiftCaseStudies->"NavigateAndLoadList" fix to match the iOS 17.

### Actual behavior

NavigationLink in example SwiftCaseStudies-> "NavigateAndLoadList" in iOS 17 has warning

### Steps to reproduce

Just open example SwiftCaseStudies->"NavigateAndLoadList"

### The Composable Architecture version information

1.7

### Destination operating system

iOS 17

### Xcode version information

lastest

### Swift Compiler version information

```shell
swift 5.9
```

**Comment by mbrandonw** [comment]
Hi @tigline, can you describe more what the issue is here? It's true that the `NavigateAndLoadList` case study is using a deprecated `NavigationLink` initializer, but the purpose of that case study is to show how to perform navigation in order versions of iOS.

Since this is not an issue with the library I am going to convert it to a discussion.

**Reply by VilemKurz** [reply]
The issue here is that apparently there is no way (or at least it is not documented) to use NavigationLink with tree based navigation on ios 16+ without resorting to deprecated api. Please, can you publish some guidance on NavigationLink with tree based navigation in doc?
